### ISSUE TITLE
[Bug] Pnpify is not patching tsserverlibrary file

### ISSUE THREAD

**Issue by AlexandreBonaventure** [yarnpkg/berry#2986]
- [x] I'd be willing to implement a fix

**Describe the bug**
The pnpify plugin is not patching the tsserverlibrary service from typescript. Some VSCode tooling like Volar uses it:
https://github.com/johnsoncodehk/volar/blob/2f8573be6034983c93bca3282672ddb7ced790f3/packages/shared/src/index.ts#L76-L81
This makes it impossible to use this tool with yarn PnP.

**Screenshots**

<img width="944" alt="Screen Shot 2021-06-04 at 12 23 08 PM" src="https://user-images.githubusercontent.com/4596409/120833316-a0fcf800-c52f-11eb-941b-8f35ba36789e.png">

**Environment if relevant (please complete the following information):**

 - Yarn version 2.4.0

**Comment by arcanis** [comment]
That's surprising - we generate the patch by comparing the output of TS' build folder before & after, so I'd expect any change to be there.

**Comment by merceyz** [comment]
`@yarnpkg/pnpify` indeed doesn't create a `typescript/lib/tsserverlibrary.js` file
https://github.com/yarnpkg/berry/blob/e53036a37f5763d2568a1f7465c9a7562e091129/packages/yarnpkg-pnpify/sources/sdks/base.ts#L170-L172

**Comment by AlexandreBonaventure** [comment]
Sorry, maybe I explained it incorrectly... 
Volar looks for a `tsserverlibrary.js` file in the sdk folder and fallback to the VSCode typescript (which is not patched to work with PnP). However, if I try to create it manually by forwarding the TS file like this, it still does not work
<img width="1926" alt="Screen Shot 2021-06-04 at 12 38 55 PM" src="https://user-images.githubusercontent.com/4596409/120835061-d73b7700-c531-11eb-864e-70fa1bd459c4.png">

NB: I tried with latest yarn 2.4 and even 3.0 RC

**Comment by AlexandreBonaventure** [comment]
(btw you guys are just so fast to provide support/answer, this is really appreciated. You rock!)

**Comment by merceyz** [comment]
Could you follow the issue template and provide a reproduction so we can test this?

**Comment by AlexandreBonaventure** [comment]
@merceyz certainly, now my question is what do you want to test exactly ? I can easily build a repro to assert the presence of the file in the SDK folder. However, patching the tsserverlibrary seems more involved than forwarding the TS module, as for tsserver you probably have to hook into the resolving logic, and I have zero knowledge about the API whatsoever...

**Comment by merceyz** [comment]
A repo using Volar that I can clone it and see the issue in your screenshot would do and/or the steps to test it

**Comment by AlexandreBonaventure** [comment]
@merceyz https://github.com/AlexandreBonaventure/yarn-pnpify-volar
If you clone this one and open in VSCode, it will recommend installing Volar extension. From then on, you can open any .vue file and the error is going to pop out

**Comment by harrytran998** [comment]
+ 1 for this issue! Hope your team can consider fixing that!

**Comment by lmarianski** [comment]
Not sure if related but i'm getting this error in Yarn + Volar
```
[Error - 01:25:42] Request textDocument/foldingRange failed.
  Message: Request textDocument/foldingRange failed with message: Cannot read property 'VariableDeclaration' of undefined
  Code: -32603 
  ```

**Comment by merceyz** [comment]
That will also be fixed in #2995 @lmarianski
